### PR TITLE
Removing BouncyCastle dependencies from ingest-attachment plugin

### PR DESCRIPTION
**Pull request by masseyke**
This removes the BouncyCastle dependencies from the ingest-attachment plugin in order to reduce the artifact size and reduce our exposure to security problems. The BouncyCastle libraries are only needed for things that we do not support anyway (decrypting PDFs for example).

### CODE REVIEW

**Comment by masseyke** [comment]
@elasticsearchmachine run elasticsearch-ci/part-1

File: plugins/ingest-attachment/build.gradle [hunk 1]
```diff
@@ -51,9 +51,6 @@ dependencies {
   api "org.apache.pdfbox:fontbox:${versions.pdfbox}"
   api "org.apache.pdfbox:jempbox:1.8.16"
   api "commons-logging:commons-logging:${versions.commonslogging}"
-  api "org.bouncycastle:bcmail-jdk15on:${versions.bouncycastle}"
-  api "org.bouncycastle:bcprov-jdk15on:${versions.bouncycastle}"
-  api "org.bouncycastle:bcpkix-jdk15on:${versions.bouncycastle}"
```
[the remark this answers is not in the document]
Comment (masseyke) [reply plugins/ingest-attachment/build.gradle:56]: They don't show up in the ingest-attachment zip file (and they previously did). So I assumed that it worked, right?

[the remark this answers is not in the document]
Comment (masseyke) [reply plugins/ingest-attachment/build.gradle:56]: A snapshot before my changes:
```
keiths-mbp:elasticsearch kmassey$ unzip -l plugins/ingest-attachment//build/distributions/ingest-attachment-8.3.0-SNAPSHOT.zip  | grep bc
   109808  07-12-2021 17:31   bcmail-jdk15on-1.64.jar
  4771726  07-12-2021 17:31   bcprov-jdk15on-1.64.jar
   878038  07-12-2021 17:31   bcpkix-jdk15on-1.64.jar
```
vs the snapshot with my changes:
```
unzip -l plugins/ingest-attachment//build/distributions/ingest-attachment-8.4.0-SNAPSHOT.zip  | grep bc
keiths-mbp:elasticsearch kmassey$ 
```

[the remark this answers is not in the document]
Comment (dakrone) [reply plugins/ingest-attachment/build.gradle:56]: Okay, that seems like good enough for me then!

**Review by dakrone — APPROVED** [review]
LGTM, thanks for doing this Keith

**Pull request merged**